Content.Client/Holopad/HolopadSystem.cs

```diff
@@ -19,20 +21,19 @@ public override void Initialize()
     {
         base.Initialize();
 
-        SubscribeLocalEvent<HolopadHologramComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<HolopadHologramComponent, ComponentHandleState>(HandleHolopadHologramState);
         SubscribeLocalEvent<HolopadHologramComponent, BeforePostShaderRenderEvent>(OnShaderRender);
         SubscribeAllEvent<TypingChangedEvent>(OnTypingChanged);
-
-        SubscribeNetworkEvent<PlayerSpriteStateRequest>(OnPlayerSpriteStateRequest);
-        SubscribeNetworkEvent<PlayerSpriteStateMessage>(OnPlayerSpriteStateMessage);
     }
 
-    private void OnComponentInit(EntityUid uid, HolopadHologramComponent component, ComponentInit ev)
+    private void HandleHolopadHologramState(Entity<HolopadHologramComponent> entity, ref ComponentHandleState args)
     {
-        if (!TryComp<SpriteComponent>(uid, out var sprite))
+        if (args.Current is not HolopadHologramComponentState state)
             return;
 
-        UpdateHologramSprite(uid);
+        entity.Comp.LinkedEntity = GetEntity(state.Target);
+
```

Comment:
Use EnsureEntity.

The difference is that EnsureEntity will re-run this state until the client actually receives the target entity (they may not arrive on the same tick due to streaming).